### PR TITLE
HPCC-17758 Standing up new LDAP OU, users cannot submit workunits

### PR DESCRIPTION
**Pull request by RussWhitehead**
When you create a new OU=Workunits, it is only granting WU scope permission
to Administrators. This means users cannot submit workunits, they get a
workunit scope error. This PR grants Authenticated Users access to this
OU so the default behavior will be to allow all HPCC users to execute workunits

Signed-off-by: Russ Whitehead <william.whitehead@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [x] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [x] This change fixes the problem, not just the symptom
  - [x] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browers
  - [ ] The component(s) render as expected

### CODE REVIEW

**Comment by RussWhitehead** [comment]
@rpastrana Please review

**Review by rpastrana — COMMENTED** [review]
Based on offline conversation with @RussWhitehead, the code change seems to do what was intended. However, I can't verify that the new security rule is acceptable.

File: system/security/LdapSecurity/ldapconnection.cpp [hunk 1]
```diff
@@ -1431,7 +1431,7 @@ class CLdapClient : implements ILdapClient, public CInterface
             createLdapBasedn(NULL, m_ldapconfig->getResourceBasedn(RT_DEFAULT), PT_ADMINISTRATORS_ONLY);
             createLdapBasedn(NULL, m_ldapconfig->getResourceBasedn(RT_FILE_SCOPE), PT_ADMINISTRATORS_ONLY);
             createLdapBasedn(NULL, m_ldapconfig->getResourceBasedn(RT_VIEW_SCOPE), PT_ADMINISTRATORS_ONLY);
-            createLdapBasedn(NULL, m_ldapconfig->getResourceBasedn(RT_WORKUNIT_SCOPE), PT_ADMINISTRATORS_ONLY);
+            createLdapBasedn(NULL, m_ldapconfig->getResourceBasedn(RT_WORKUNIT_SCOPE), PT_DEFAULT);//Include "Authenticated Users" to enable default behavior
```
Comment (richardkchapman) [inline system/security/LdapSecurity/ldapconnection.cpp]: Why would this be any different from the other scopes?

Feels safer to default to no rights when people are setting up LDAP to me...

Comment (RussWhitehead) [inline system/security/LdapSecurity/ldapconnection.cpp]: This scope is different from the others in that it is the only one with a "default" setting to be applied when there are no scopes defined for a given user. By granting access to authenticated users (the default behavior), users will be able to run workunits out of the box.  The alternative is that an admin would have to create a scope for each and every user.
That being said, our LDAP Admins have stated that we cannot grant Authenticated Users access to any OU, so I am cancelling this PR and will investigate an alternative approach

**Pull request closed without merging**